### PR TITLE
doxygen: refine grouping of kernel docs

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

Some refinement and restructuring the doxygen documentation of the kernel.

1. fix grouping of sync mechanisms
2. adapt using `@ingroup` instead of `@addtogroup`

### Testing procedure

build the docs and check the grouping in `Modules->Kernel` specifically look into the sync branch.

### Issues/PRs references

None

### CODE REVIEW

File: core/doc.txt [hunk 1]
```diff
@@ -25,3 +25,9 @@
  * @ingroup     core
  * @brief       Configuration data and startup code for the kernel
  */
+
+ /**
+  * @defgroup    core_sync Synchronization
```
Comment (aabadie) [inline core/doc.txt]: Maybe call the group `Thread Synchronization` ?

Comment (smlng) [reply core/doc.txt]: sounds good

**Comment by smlng** [comment]
@aabadie addressed comments, good to squash?

**Comment by smlng** [comment]
done!

**Comment by jia200x** [comment]
Merge? :)

**Comment by aabadie** [comment]
Thanks for the reminder @jia200x, I missed it this morning!

**Pull request merged**